webrtc.html

```diff
@@ -3023,27 +3021,48 @@ <h2>Methods</h2>
                 adopt the pending local description, or rollback to the current
                 description if the remote side rejected the change.</p>
                 <p>As noted in <span data-jsep="modifying-sdp">[[!JSEP]]</span>
-                the SDP returned from <code>createOffer</code> or
-                <code>createAnswer</code> MUST NOT be changed
-                before passing it to <code>setLocalDescription</code>. As
-                a result, when the method is invoked, the user agent MUST
-                run the following steps:</p>
+                the SDP passed to <code>setLocalDescription</code> is not
+                allowed to have changed from when it was returned from either
+                <code>createOffer</code> or <code>createAnswer</code>.</p>
+                <p>When the method is invoked, the user agent MUST run the
+                following steps:</p>
                 <ol>
-                  <li>Let <var>description</var> be the first argument to
-                  <code>setLocalDescription</code>.</li>
-                  <li data-tests="RTCPeerConnection-setLocalDescription-answer.html">If <code><var>description</var>.sdp</code> is the
-                  empty string and <code><var>description</var>.type</code>
-                  is <code>"answer"</code> or <code>"pranswer"</code>,
-                  set <code><var>description</var>.sdp</code> to the value of
-                  <var>connection</var>'s <a>[[\LastCreatedAnswer]]</a> slot.</li>
-                  <li data-tests="RTCPeerConnection-setLocalDescription-offer.html">If <code><var>description</var>.sdp</code> is the
-                  empty string and <code><var>description</var>.type</code>
-                  is <code>"offer"</code>, set <code><var>description</var>.sdp</code>
-                  to the value of <var>connection</var>'s
-                  <a>[[\LastCreatedOffer]]</a> slot.</li>
-                  <li>Return the result of <a href="#set-description">setting
-                  the RTCSessionDescription</a> indicated by
-                  <code><var>description</var></code>.</li>
+                  <li>
+                    <p>Let <var>description</var> be the method's first
+                    argument.</p>
+                  </li>
+                  <li>
+                    <p>Let <var>connection</var> be the
+                    <code><a>RTCPeerConnection</a></code> object on which the
+                    method was invoked.</p>
+                  </li>
+                  <li>
+                    <p>Return the result of <a href=
+                    "#enqueue-an-operation">enqueuing</a> the following steps to
+                    <var>connection</var>'s operation queue:</p>
+                    <ol>
```

Comment:
We are enqueuing a task that sets the session description (= creates a promise and then enqueues a task to resolve it). It doesn't make sense to me to enqueue a task to enqueue a task.
1. It adds an additional delay, that sounds like a bug to me. While it probably rarely creates problems, it doesn't seem to add any value to do so.
2. Language-wise, it doesn't make sense to return the result of enqueuing an operation. The promise we want to return hasn't been created at this point. We would have to create another promise and wrap the other promise in it. But see 1), we don't want to do that.

Reply:
Resolved offline. [Enqueueing an operation](http://w3c.github.io/webrtc-pc/#enqueue-an-operation) is not the same as [queue a task](https://html.spec.whatwg.org/multipage/webappapis.html#queue-a-task). The former chains an asynchronous operation on the `RTCPeerConnection`'s internal serialization queue. The latter takes us back to main thread, adding an event to the JS event loop.

Reply:
The former also returns a promise, so we should be good here.